### ISSUE TITLE
CSGShapes created at runtime have unexpected collision behaviour

### ISSUE THREAD

**Issue by SgtLion** [godotengine/godot#41653]
**Godot version:**
v3.2.2.stable.official

**OS/device including version:**
Windows 10 1909

**Issue description:**
I have a project that is generating prototype levels by creating sets of CSGBoxes and translating/rotating them accordingly, they are created by doing CSGBox.new() and add_child().

When created by code as above, they look visually correct, but collision shapes and invisible walls are all over the show, some part of translation/rotation isn't working correctly, I think, but it's not consistent. If the scene is then saved as a .tscn and recreated, it then works perfectly as expected. Nothing strange shows up when debugging and looking at the remote scene.

**Steps to reproduce:**
Create a 3D scene with a moveable character. Create a script for the top node that in _ready(), creates basic floors/walls with CSGBox.new() and set translations appropriately (and enable use_collision). Invisible walls will stop the character moving around, and the CSGBoxes will likely be collisionless.

For extra proof, save the scene by packing it:

```
var saved_scene = PackedScene.new()
saved_scene.pack(get_tree().get_current_scene())
ResourceSaver.save("res://savedscene.tscn", saved_scene)
```

Then load it from the editor, and watch it suddenly work perfectly.

**Minimal reproduction project:**
[HuntDownTheWall.zip](https://github.com/godotengine/godot/files/5151841/HuntDownTheWall.zip)

runtime_generated_scene.tscn will generate the CSGBoxes at runtime. You will find you cannot run all the way around the box because of invisible walls.

identical_pregenerated_scene.tscn is a perfect replica of the generated scene, you will find this one works perfectly as expected.

**Comment by Calinou** [comment]
cc @madmiraal

**Comment by hoontee** [comment]
Duplicate of #40683 / #27364.

**Comment by aaronfranke** [comment]
@SgtLion Are you able to reproduce this issue in the latest master branch? See [this article](https://docs.godotengine.org/en/latest/development/compiling/index.html) or grab a build [here](https://ifiregames.itch.io/stern-flowers-chibifire-com-godot-engine) or [here](https://hugo.pro/projects/godot-builds/).

**Comment by hoontee** [comment]
@aaronfranke The issue isn't reproducible in `master` after fc8574fbac1073f4d68197dc8a2b3504bfc8970d:
![](https://i.gyazo.com/7ee04d3a4f470c05b2ac40713c0513d8.gif)
[The PR behind the change](40935) has yet to be cherry-picked for 3.2.

**Comment by madmiraal** [comment]
Closing as duplicate, fixed in #40935.
